src/Symfony/Component/Validator/Constraints/BackedEnumValue.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+
+/**
+ * Validates that a value is one of an enum.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+class BackedEnumValue extends Constraint
+{
+    public const NO_SUCH_CHOICE_ERROR = '53dcc1b1-a8dd-4813-baa5-b8486ff56447';
+    public const INVALID_TYPE_ERROR = 'aa0374f4-b3ab-4362-b48d-b5ecf0f1a02d';
+
+    protected const ERROR_NAMES = [
+        self::NO_SUCH_CHOICE_ERROR => 'NO_SUCH_CHOICE_ERROR',
+        self::INVALID_TYPE_ERROR => 'INVALID_TYPE_ERROR',
+    ];
+
+    /**
+     * @param class-string<\BackedEnum> $type   the type of the enum
+     * @param \BackedEnum[]             $only   the cases that should be considered valid
+     * @param \BackedEnum[]             $except the cases that should be considered invalid
+     */
+    public function __construct(
+        public string $type,
+        public array $only = [],
+        public array $except = [],
+        public string $message = 'The value you selected is not a valid choice.',
+        public string $typeMessage = 'This value should be of type {{ type }}.',
+        ?array $groups = null,
+        mixed $payload = null,
+    ) {
+        parent::__construct([], $groups, $payload);
+
+        if (\count($only) > 0 && \count($except) > 0) {
+            throw new ConstraintDefinitionException(sprintf('The option "only" and "except" can not be applied in the same constraint "%s".', __CLASS__));
+        }
+
+        foreach ($only as $onlyValue) {
+            if (!is_a($onlyValue, \BackedEnum::class, true)) {
+                throw new ConstraintDefinitionException('The "only" values must be a BackedEnum.');
+            }
+        }
```

Comment:
Why do you validate `$only` and `$exception` inside the constructor, but `$type` inside the validator?